stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/attachment/UploadAttachmentsWorker.kt

```diff
@@ -1,19 +1,40 @@
 package io.getstream.chat.android.offline.message.attachment
 
 import android.content.Context
+import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.errors.ChatError
 import io.getstream.chat.android.client.models.Attachment
 import io.getstream.chat.android.client.utils.Result
-import io.getstream.chat.android.offline.ChatDomain
 import io.getstream.chat.android.offline.ChatDomainImpl
 
 internal class UploadAttachmentsWorker(private val appContext: Context) {
-    suspend fun uploadAttachmentsForMessage(channelType: String, channelId: String, messageId: String): Result<Unit> {
+    suspend fun uploadAttachmentsForMessage(
+        channelType: String,
+        channelId: String,
+        messageId: String,
+        chatDomain: ChatDomainImpl,
+        chatClient: ChatClient
+    ): Result<Unit> {
         return try {
-            val domainImpl = (ChatDomain.instance() as ChatDomainImpl)
+            val domainImpl = chatDomain.apply {
+                if (!isRepositoryInitialized) {
+                    chatClient.setUserWithoutConnectingIfNeeded()
+
+                    chatClient.getCurrentUser()?.let(::initRepositoryFacade)
```

Comment:
I'm not sure if this will work correctly. `chatClient.setUserWithoutConnectingIfNeeded()` sets the user,  then notifies `ChatDomain` which does the whole initialization. Why do we need to do it manually?

Reply:
You're right, no need to do it manually.